### PR TITLE
Ele 1668 improve accuracy by combining z score with another mechanism

### PR DESCRIPTION
**Pull request by dapollak**

### CODE REVIEW

File: integration_tests/tests/test_volume_anomalies.py [hunk 1]
```diff
@@ -204,3 +204,69 @@ def test_wildcard_name_table_volume_anomalies(test_id: str, dbt_project: DbtProj
         table_name=test_id,
     )
     assert test_result["status"] == "fail"
+
+@Parametrization.autodetect_parameters()
+@Parametrization.case(name="true_positive", expected_result="fail", drop_mean_percent_deviation=5, metric_value=25)
+@Parametrization.case(name="false_positve", expected_result="fail", drop_mean_percent_deviation=None, metric_value=29)
```
Comment (haritamar) [inline integration_tests/tests/test_volume_anomalies.py:210]:
```suggestion
@Parametrization.case(name="false_positive", expected_result="fail", drop_mean_percent_deviation=None, metric_value=29)
```

File: integration_tests/tests/test_volume_anomalies.py [hunk 2]
```diff
@@ -204,3 +204,69 @@ def test_wildcard_name_table_volume_anomalies(test_id: str, dbt_project: DbtProj
...
+        **DBT_TEST_ARGS,
+        "time_bucket": {"period": "day", "count": 1},
+        "drop_mean_percent_deviation": drop_mean_percent_deviation
+    }
+    test_result = dbt_project.test(test_id, DBT_TEST_NAME, test_args, data=data)
+    assert test_result["status"] == expected_result
+
+@Parametrization.autodetect_parameters()
+@Parametrization.case(name="true_positive", expected_result="fail", spike_mean_percent_deviation=5, metric_value=35)
+@Parametrization.case(name="false_positve", expected_result="fail", spike_mean_percent_deviation=None, metric_value=31)
```
Comment (haritamar) [inline integration_tests/tests/test_volume_anomalies.py:243]:
```suggestion
@Parametrization.case(name="false_positive", expected_result="fail", spike_mean_percent_deviation=None, metric_value=31)
```

File: macros/edr/tests/test_utils/get_anomaly_query.sql [hunk 3]
```diff
@@ -106,3 +111,36 @@
...
+        {% if drop_mean_percent_deviation and spike_mean_percent_deviation %}
+            ({{ spike_query }} or {{ drop_query }})
+        {% elif not drop_mean_percent_deviation and spike_mean_percent_deviation %}
+            {{ spike_query }}
+        {% elif not spike_mean_percent_deviation and drop_mean_percent_deviation %}
+            {{ drop_query }}
+        {% else %}
+            (1 = 1)
+        {% endif %}
+    {% endif %}
```
Comment (haritamar) [inline macros/edr/tests/test_utils/get_anomaly_query.sql:145]: Maybe this macro can be simplified to:
```
{% spike_mean_percent_deviation and anomaly_direction | lower in ['spike', 'both'] %}
    (metric_value > ((1 + {{ spike_mean_percent_deviation }}/100.0) * training_avg))
{% else %}
    (1 = 1)
{% endif %}

and

{% drop_mean_percent_deviation and anomaly_direction | lower in ['drop', 'both'] %}
    (metric_value < ((1 - {{ drop_mean_percent_deviation }}/100.0) * training_avg))
{% else %}
    (1 = 1)
{% endif %}
```
?

File: macros/edr/tests/test_configuration/get_anomalies_test_configuration.sql [hunk 4]
```diff
@@ -34,6 +36,9 @@
     {%- set days_back = elementary.get_days_back(days_back, model_graph_node, seasonality) %}
     {%- set seasonality = elementary.get_seasonality(seasonality, model_graph_node, time_bucket, timestamp_column) %}
 
+    {%- set spike_mean_percent_deviation = elementary.get_test_argument('spike_mean_percent_deviation', spike_mean_percent_deviation, model_graph_node) %}
+    {%- set drop_mean_percent_deviation = elementary.get_test_argument('drop_mean_percent_deviation', drop_mean_percent_deviation, model_graph_node) %}
```
[the remark this answers is not in the document]
Comment (haritamar) [reply macros/edr/tests/test_configuration/get_anomalies_test_configuration.sql:40]: Also maybe fail if one of these is not compatible with the anomaly direction

File: macros/edr/tests/test_utils/get_anomaly_query.sql (hunk 3 above)
Comment (dapollak) [reply macros/edr/tests/test_utils/get_anomaly_query.sql]: yup

**Review by haritamar — APPROVED** [review]

**Pull request merged**